uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/app/IApp.java

```diff
@@ -0,0 +1,55 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.app;
+
+public interface IApp {
+
+    /**
+     * Install this app on the device
+     */
+    void install();
+
+    /**
+     * Launch this app on the device
+     */
+    void launch();
+
+    /**
+     * Clear (storage) associated to this app on the device
+     */
+    void clear();
+
+    /**
+     * Remove this app from the device
+     */
+    void uninstall();
+
+    /**
+     * Handle the first run experience for this app on first time launch
+     */
+    void handleFirstRun();
+
+    boolean hasPermission(String permission);
```

Comment:
javadoc here too -- are these android permissions? something else? not clear at a glance

Reply:
Addressed